virtio-devices/src/console.rs

```diff
@@ -332,6 +473,46 @@ impl EpollHelperHandler for ConsoleEpollHandler {
         }
         Ok(())
     }
+
+    fn handle_timeout(&mut self, helper: &mut EpollHelper) -> Result<(), EpollHelperError> {
+        if !self.endpoint.is_pty() {
+            return Ok(());
+        }
+
+        if self.file_event_registered {
+            // This very specific case happens when the console is connected
+            // to a PTY. We know EPOLLHUP is always present when there's nothing
+            // connected at the other end of the PTY. That's why getting no event
+            // means we can flush the output of the console through the PTY.
+            self.trigger_pty_flush()
+                .map_err(EpollHelperError::HandleTimeout)?;
+        }
+
+        self.register_file_event(helper)
```

Comment:
It is good to have some comments here. My understanding is that this is for re-checking if a pty is connected every 500ms (the epoll timeout).

Reply:
I'll add a comment.